Servers/UI/OJS.Servers.Ui/Controllers/Api/SubmissionsController.cs

```diff
@@ -74,37 +60,50 @@ public IActionResult Download(int id)
     }
 
     /// <summary>
-    /// Gets a subset of submissions by specific problem and given take count.
+    /// Gets a subset of submissions by specific problem.
     /// </summary>
     /// <param name="id">The id of the problem.</param>
     /// <param name="isOfficial">Should the submissions be only from compete mode.</param>
-    /// <param name="take">Number of submissions to return.</param>
+    /// <param name="page">Current submissions page.</param>
     /// <returns>A collection of submissions for a specific problem.</returns>
     [HttpGet("{id:int}")]
-    [ProducesResponseType(typeof(IEnumerable<SubmissionResultsResponseModel>), Status200OK)]
+    [ProducesResponseType(typeof(PagedResultResponse<SubmissionResultsResponseModel>), Status200OK)]
     public async Task<IActionResult> GetSubmissionResultsByProblem(
         int id,
         [FromQuery] bool isOfficial,
-        [FromQuery] int take)
+        [FromQuery] int page)
         => await this.submissionsBusiness
-            .GetSubmissionResultsByProblem(id, isOfficial, take)
-            .MapCollection<SubmissionViewInResultsResponseModel>()
+            .GetSubmissionResultsByProblem(id, isOfficial, page)
+            .Map<PagedResultResponse<SubmissionResultsResponseModel>>()
             .ToOkResult();
 
     /// <summary>
-    /// Gets a subset of submission results and details  for the selected user by specific problem and given take count.
+    /// Gets a subset of submission results for the selected user by specific problem in details page.
     /// </summary>
     /// <param name="submissionId">The id of the submission.</param>
-    /// <param name="take">Number of submissions to return.</param>
+    /// <param name="page">Current submissions page.</param>
     /// <returns>A collection of submissions for a specific problem.</returns>
     [HttpGet("{submissionId:int}")]
-    [ProducesResponseType(typeof(IEnumerable<SubmissionResultsResponseModel>), Status200OK)]
-    public async Task<IActionResult> GetSubmissionDetailsWithResults(
+    [ProducesResponseType(typeof(PagedResultResponse<SubmissionResultsResponseModel>), Status200OK)]
+    public async Task<IActionResult> GetSubmissionResults(
         int submissionId,
-        [FromQuery] int take)
+        [FromQuery] int page)
         => await this.submissionsBusiness
-            .GetSubmissionDetailsWithResults(submissionId, take)
-            .Map<SubmissionDetailsWIthResultsResponseModel>()
```

Comment:
Why removed the old logic ? With the new implementation it will create 2 requests on page load.